packages/eas-cli/src/utils/easJson/EasJsonAccessorWithCustomErrors.ts

```diff
@@ -0,0 +1,38 @@
+import { EasJson, EasJsonAccessor } from '@expo/eas-json';
+import { InvalidEasJsonError } from '@expo/eas-json/build/errors';
+import chalk from 'chalk';
+import { ValidationError } from 'joi';
+
+import Log, { link } from '../../log';
+
+export const easJsonCustomErrorMessageHandlers: ((err: ValidationError) => void)[] = [
+  // Ask user to upgrade eas-cli version or check the docs when image is invalid.
+  (err: ValidationError) => {
+    for (const detail of err.details) {
+      if (detail.path[detail.path.length - 1] === 'image') {
+        Log.error(
+          `The image you specified in ${chalk.bold(
+            'eas.json'
+          )} is invalid for this EAS CLI version. Please check if your EAS CLI version is up to date. Othervise visit ${link(
+            'https://docs.expo.dev/build-reference/infrastructure/'
+          )} to learn more about supported images.`
+        );
```

Comment:
@brentvatne @kbrandwijk Is this message correct?

Reply:
'invalid for this EAS CLI version' sounds a bit cryptic. 



'Specified build image 'xyz' is not recognized. Please update your EAS CLI and see '...' for supported build images'.



Maybe something like this, which I'm also not 100% happy with, so I'll think about it for a bit.

Reply:
Also, can we swallow the server response, because the list of images returned is actually the outdated list from the current eas-cli version eas.json schema definition, which adds to the confusion.

Reply:
> Also, can we swallow the server response, because the list of images returned is actually the outdated list from the current eas-cli version eas.json schema definition, which adds to the confusion.

@kbrandwijk Sorry, I'm not sure if I understand you correctly - so you are suggesting to not validate the build image based on the offline schema specification attached to every CLI version, but instead, just show the server error we get once submitting the build with an incorrect image? Is this correct?

Reply:
I think Kim wanted to only print custom errors without the original Joi error.

Reply:
Ah right, yes that is what I meant.